script/lint.js

```diff
@@ -136,12 +137,54 @@ const LINTERS = [ {
       process.exit(1)
     }
   }
+}, {
+  key: 'patches',
+  roots: ['patches'],
+  test: () => true,
+  run: () => {
+    const patchesDir = path.resolve(__dirname, '../patches')
+    for (const patchTarget of fs.readdirSync(patchesDir)) {
+      const targetDir = path.resolve(patchesDir, patchTarget)
+      // If the config does not exist that is OK, we just skip this dir
+      const targetConfig = path.resolve(targetDir, 'config.json')
+      if (!fs.existsSync(targetConfig)) continue
+
+      const config = JSON.parse(fs.readFileSync(targetConfig, 'utf8'))
+      for (const key of Object.keys(config)) {
+        // The directory the config points to should exist
+        const targetPatchesDir = path.resolve(__dirname, '../../..', key)
+        expect(fs.existsSync(targetPatchesDir)).to.equal(true, `target patch directory: "${targetPatchesDir}" should exist`)
+        // We need a .patches file
+        const dotPatchesPath = path.resolve(targetPatchesDir, '.patches')
+        expect(fs.existsSync(dotPatchesPath)).to.equal(true, `.patches file: "${dotPatchesPath}" should exist`)
+
+        // Read the patch list
+        const patchFileList = fs.readFileSync(dotPatchesPath, 'utf8').trim().split('\n')
+        const patchFileSet = new Set(patchFileList)
+        expect(patchFileList).to.have.length(patchFileSet.size, 'each patch file should only be in the .patches file once')
```

Comment:
Should list the name that's shown twice, e.g.
```suggestion
        const duplicates = (list) => {
          const count = new Map
          for (const el of list) {
            count.set(el, (count.get(el) || 0) + 1)
          }
          return [...count.entries()].filter(([k, v]) => v > 1).map(([k, v]) => k)
        }
        const duplicatedPatches = duplicates(patchFileList)
        if (duplicatedPatches.length > 0) {
          throw new Error(`Patches appear in the list more than once: ${duplicatedPatches}`)
        }
```

or

```suggestion
        patchFileList.reduce((seen, file) => {
          if (seen.has(file)) {
            throw new Error(`'${file}' is listed in ${dotPatchesPath} more than once`)
          }
          return seen.add(file)
        }, new Set)
```